### ISSUE TITLE
Use Dec and I128 as default numeric types

### ISSUE THREAD

**Issue by rtfeldman** [roc-lang/roc#2847]
Although `Dec` isn't feature-complete yet, it's far enough that we should start using it as the default fractional type. So putting `0.1 + 0.2` into the repl should return `0.3`.

Similarly, we should start using `I128` instead of `I64` as both the default integer type and also the default numeric type. It's the same size as `Dec` and minimizes the chance that throwing calculations into the repl will overflow. That seems like the situation where numbers with no types specified is most likely to come up, and it's not performance-critical, so using the slower `I128` seems fine.

Basically what needs to happen here is that when when we do code generation for a number, if it's...
* `Num *` then we should treat it as an `I128`
* `Int *` then we should treat it as an `I128`
* `Float *` then we should treat it as a `Dec`

(Currently we use `I64` and `F64` for these instead.)
